src/Dialect/ONNX/ONNXOps/Quantize/DequantizeLinear.cpp

```diff
@@ -34,6 +34,50 @@ int64_t nonScalar1DLen(ShapedType ty) {
 
 } // namespace
 
+namespace onnx_mlir {
+
+template <>
+LogicalResult ONNXDequantizeLinearOpShapeHelper::computeShape() {
+  ONNXDequantizeLinearOpAdaptor operandAdaptor(
+      operands, op->getAttrDictionary());
+  RankedTensorType xTy =
+      operandAdaptor.x().getType().dyn_cast<RankedTensorType>();
+  DimsExpr outputDims;
+  createIE->getShapeAsDims(operandAdaptor.x(), outputDims);
+
+  // Get d.
+  int64_t d =
+      nonScalar1DLen(operandAdaptor.x_scale().getType().cast<ShapedType>());
+  if (d == -1 && !isFromNone(operandAdaptor.x_zero_point())) {
+    d = nonScalar1DLen(
+        operandAdaptor.x_zero_point().getType().cast<ShapedType>());
+  }
+
+  if (d != -1) {
+    int64_t r = xTy.getRank();
+    int64_t a = operandAdaptor.axis();
+    // Checked in verify:
+    assert(-r <= a && a < r && "axis out of range");
+    if (a < 0)
+      a += r;
+    if (!outputDims[a].isLiteral()) {
+      outputDims[a] = LiteralIndexExpr(d);
+    } else {
```

Comment:
Looks like this `else` is redundant. `assert` can come without `else`.

Reply:
I see, fine I can remove the else.